crates/bevy_ecs/src/schedule/commands.rs

```diff
@@ -0,0 +1,126 @@
+use super::{Schedule, StageLabel, SystemDescriptor};
+use crate::system::Command;
+use crate::world::World;
+
+#[derive(Default)]
+pub struct SchedulerCommandQueue {
+    items: Vec<Box<dyn SchedulerCommand>>,
+}
+
+impl SchedulerCommandQueue {
+    pub fn push<C>(&mut self, command: C)
+    where
+        C: SchedulerCommand,
+    {
+        self.items.push(Box::new(command));
+    }
+
+    pub fn apply(&mut self, schedule: &mut Schedule) {
+        for command in self.items.drain(..) {
+            command.write(schedule);
+        }
+    }
+
+    pub fn transfer(&mut self, queue: &mut SchedulerCommandQueue, offset: usize) {
```

Comment:
This signature doesn't feel immediately clear to me. What transfers into what here?

I think we can mimic `Vec::drain()` - sure, it does more than we need, but it's tapping into design of Rust's own APIs. No need to create your own iterator, either: just return the one you get from `Vec::drain()`.

---

Scratch that, a better note is below.